source/auth/tests/README.rst

```diff
@@ -0,0 +1,55 @@
+==========
+Auth Tests
+==========
+
+The YAML and JSON files in this directory tree are platform-independent tests
+that drivers can use to prove their conformance to the Auth Spec at least with
+respect to connection string URI input.
+
+Drivers should do additional unit testing if there are alternate ways of
+configuring credentials on a client.
+
+Driver must also conduct the prose tests in the Auth Spec test plan section.
+
+Format
+------
+
+Each YAML file contains an object with a single ``tests`` key. This key is an
+array of test case objects, each of which have the following keys:
+
+- ``description``: A string describing the test.
+- ``uri``: A string containing the URI to be parsed.
+- ``valid:`` A boolean indicating if the URI should be considered valid.
+- ``credential_configured:`` A boolean indicating if the URI is considered to have
+  configured a credential for the purposes of deciding if the driver should
+  authenticate to the topology.
+- ``credential``: An object containing the following keys:
+
+  - ``username``: A string containing the username. For auth mechanisms
+    that do not utilize a password, this may be the entire ``userinfo`` token
+    from the connection string.  If this is the empty string,
+    test harnesses should interpret it as unspecified.
+  - ``password``: A string containing the password.  If this is the empty string,
+    test harnesses should interpret it as unspecified.
+  - ``source``: A string containing the authentication database.
+  - ``mechanism``: A string containing the authentication mechanism.  The string "default"
```

Comment:
I prefer null to a magic string for non-associating values.

Reply:
Please resolve this one too.  Now that we're using null to indicate that a value is not set, this should be `mechanism: null` instead of `mechanism: "default"`

Reply:
Will do.